controllers/drplacementcontrol_controller.go

```diff
@@ -816,6 +857,30 @@ func (r *DRPlacementControlReconciler) clonePlacementRule(ctx context.Context,
 	return clonedPlRule, nil
 }
 
+func (r *DRPlacementControlReconciler) getVRGsFromManagedClusters(drpc *rmn.DRPlacementControl,
```

Comment:
This is not a review comment, it is more a statement of understanding:

### `deploy` ONLY action
- During `deploy` action, we would invoke this to find actual VRG state on the clusters, and then proceed as follows:
    - If a cluster is down, we may not get a response, and we will requeue reconciliation
    - If a stale status is reported, from a prior MCV output before the cluster went down, it can be reported back as P(rimary)/S(condary)
        - If stale status is P and cluster is down, we proceed normally (we have a primary and will reconcile this as the current cluster later)
        - If stale status is S and VRG as Secondary reports required success conditions, we can proceed; else if VRG status as Secondary remains unknown, we will requeque reconciliation (can happen if older action was relocate and we marked older cluster as S and were waiting for it to report secondary, at which point user changes spec to remove the action)
    - If both report P (due to stale status on one end and current status on the other, can happen if a prior failover action was invoked and user removed action in between reconcile), report errors and try again (failover should have marked VRG MW on down cluster as S, before marking VRG MW as P on the surviving cluster (which is when MCV will find a primary everywhere), so at some point in time when the cluster recovers, we would be able reconcile this further)

Reply:
Rehashing deploy action understanding (in the current scheme):
- Ensure homeCluster either has VRG as Primary, or does not have a VRG
- Ensure there are no VRGs in ANY state in peerClusters
- If both conditions are met, proceed with deployment